app/domain/authentication/authn_oidc/authenticate.rb

```diff
@@ -59,8 +63,8 @@ def new_token(input)
         )
       end
 
-      def conjur_username(id_token, id_token_username_field)
-        conjur_username = id_token[id_token_username_field]
+      def conjur_username(id_token_attributes, id_token_username_field)
```

Comment:
Note: You should never have to pass parameters around like when using a `CommandClass`.  It's explicitly designed to relieve you of this burden (I should document this specifically in the gem docs, in fact).  

All of our inputs are available as member variables.  And anything *derived* from our inputs can be made available as a (memoized, if needed) helper method.

You need not worry about concurrency or multiple requests calling `call` simultaneously.  Under the hood, every time `call` is invoked a new private `Call` object is created.  The input member variables exist on *that* object.  So it is completely thread safe.